rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -616,11 +608,9 @@ private void legacyProcessPegin(
             default:
                 markTxAsProcessed(btcTx);
                 String message = String.format("Invalid peg-in protocol version: %d", protocolVersion);
-                logger.warn("[legacyProcessPegin] {}", message);
+                logger.warn("[legacyRegisterPegin] {}", message);
                 throw new RegisterBtcTransactionException(message);
         }
-
-        markTxAsProcessed(btcTx);
```

Comment:
This was used and needed for the case when a peg-in v1 is [surpassed the locking cap](https://github.com/rsksmart/rskj/blob/d2bbe46c700e1c141109e99c43e2b86dce1d531a/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L678) and a refund tx should be generated. Since this is no longer possible, removing it is safe, right? Should we add a comment mention this [here](https://github.com/rsksmart/rskj/blob/d2bbe46c700e1c141109e99c43e2b86dce1d531a/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L678) at least?

Reply:
thats a good point, i think we should call `markTxAsProcessed` right after refunding the sender

Reply:
Good catch Natha, it was marked as processed after a rejection for legacy peg-ins but not for v1 peg-ins.

Added the change. Looks like we don't have a test covering this, I'll create a new task to include this test

Reply:
Thanks!